src/tcfv2/api.ts

```diff
@@ -22,8 +25,34 @@ const api = (command: Command) =>
 		}
 	});
 
-export const getTCData = (): Promise<TCData> =>
-	api('getTCData') as Promise<TCData>;
+export const getTCData = (): Promise<TCData> => api('getTCData');
 
 export const getCustomVendorConsents = (): Promise<CustomVendorConsents> =>
-	api('getCustomVendorConsents') as Promise<CustomVendorConsents>;
+	api('getCustomVendorConsents') as unknown as Promise<CustomVendorConsents>;
+
+export const tcfApiEventListener = (): void => {
+	// https://documentation.sourcepoint.com/api/gdpr-tcf-v2-api/the-__tcfapi-gettcdata-api-overview/using-__tcfapi-addeventlistener-and-removeeventlistener-commands
+	if (window.__tcfapi) {
+		window.__tcfapi('addEventListener', 2, (result, success) => {
+			const { eventStatus } = result;
+			log('cmp', 'Tcf api event:', eventStatus);
+
+			if (!success) {
+				log('cmp', 'Tcf api addEventListener failed:');
+			}
+
+			switch (eventStatus) {
+				case 'tcloaded':
+					mark('cmp-tcfv2-got-consent');
+					invokeCallbacks();
+					break;
+				case 'useractioncomplete':
+					mark('cmp-tcfv2-user-action-complete');
```

Comment:
I'm open to other names but to me the Sourcepoint one is pretty clear! The user has completed an action associated with tfc consent.
I think you're right that it can be triggered multiple times per page load if for example the user accepts all cookies then changes their choice. Is that problematic?

Reply:
`useractioncomplete` is an [IAB term](https://github.com/InteractiveAdvertisingBureau/iabtcf-es/blob/81d39d2df895bb567a096c733f07cf5f73bac974/modules/cmpapi/src/status/EventStatus.ts)